File: 백준/Silver/2108. 통계학/통계학.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Map<Integer, Integer> frequencyMap = new HashMap<Integer, Integer>();       // {key: 숫자, value: 빈도}
        int N = Integer.parseInt(br.readLine());
        int i, inputNum;
        int avg, mid, range;
        int total = 0;

        List<Integer> numbers = new ArrayList<>();
        List<Integer> frequencyList = new ArrayList<>();

        for (i = 0; i < N; i ++) {
            inputNum = Integer.parseInt(br.readLine());
            total += inputNum;
            numbers.add(inputNum);
            if (frequencyMap.containsKey(inputNum)) {
                frequencyMap.put(inputNum, frequencyMap.get(inputNum) + 1);
            }
            else {
                frequencyMap.put(inputNum, 1);
            }
        }

        Collections.sort(numbers);

        avg = (int) Math.round(((double) total / N)); // 산술 평균 (반올림)
        mid = numbers.get(N / 2);   // 중앙 값
        range = Collections.max(numbers) - Collections.min(numbers); // 범위

        for (int key: frequencyMap.keySet()) {  // map의 key들로 반복
            if (frequencyMap.get(key) == Collections.max(frequencyMap.values())) {      // key로 가져온 map의 value(빈도)와 value(빈도)의 최대값 비교, 같으면 최빈값
                frequencyList.add(key);
            }
        }
        Collections.sort(frequencyList);

        System.out.println(avg);
        System.out.println(mid);
        if (frequencyList.size() > 1) {
            System.out.println(frequencyList.get(1));
        }
        else {
            System.out.println(frequencyList.get(0));
        }
        System.out.println(range);
    }
}

File: 백준/Silver/2108. 통계학/통계학.py
def frequency(nums):
    dictNums = {}
    for num in nums:
        if num not in dictNums:
            dictNums[num] = 1
        else:
            dictNums[num] += 1

    maxDictNums = max(dictNums.values())
    countingList = []

    for value in dictNums:
        if dictNums[value] == maxDictNums:
            countingList.append(value)

    if len(countingList) > 1:                   # 최빈값이 여러 개인 경우
        print(countingList[1])
    else:                                       # 최빈값이 하나인 경우
        print(countingList[0])

N = int(input())
nums = [int(input()) for _ in range(N)]

print(round(sum(nums) / N))           # 산술 평균

nums.sort()         
print(nums[N // 2])             # 중앙값

frequency(nums)                 # 최빈값

print(max(nums) - min(nums))    # 범위